doc/reference/javascript_cheatsheet.rst

```diff
@@ -276,4 +276,237 @@ using the view in the kanban arch (a specific example is the helpdesk dashboard)
+    * line 4: we are calling the function close on our object. it takes a function as parameter (the callback) and it happens that resolve is already a function, so we can pass it directly. To be clearer, I could have written:
+
+    .. code-block:: javascript
+
+        return new Promise (function(resolve) {
+            self.close(function() {
+                resolve();
+            });
+        });
+
+
+
+- creating a promise generator (calling one promise after the other *in sequence* and waiting for the last one)
+    Suppose that you need to loop over an array, do an operation *in sequence* and resolve a promise when the last operation is done.
+
+    .. code-block:: javascript
+
+        function doStuffOnArray(arr) {
+            var done = Promise.resolve();
+            arr.forEach(function(item) {
+                done = done.then(function() {
+                    return item.doSomethingAsynchronous();
+                });
+            });
+            return done;
+        }
+
+    This way, the promise you return is effectively the last promise.
+- creating a promise, then resolving it outside the scope of its definition (anti-pattern)
+    .. note:: we do not recommend using this, but sometimes it is useful
+
+    .. code-block:: javascript
+
+        ...
+        var resolver, rejecter;
+        var prom = new Promise(function(resolve, reject){
+            resolver = resolve;
+            rejecter = reject;
+        });
+        ...
+
+        resolver("done"); // will resolve the promise prom with the result "done"
+        rejecter("error"); // will reject the promise prom with the reason "error"
+
+Waiting for Promises
+--------------------
+- waiting for a number of Promises
+        if you have multiple promises that all need to be waited, you can convert them into a single promise that will be resolved when all the promises are resolved using Promise.all(arrayOfPromises).
+
+        .. code-block:: javascript
+
+                var prom1 = doSomethingThatReturnsAPromise();
+                var prom2 = Promise.resolve(true);
+                var constant = true;
+
+                var all = Promise.all([prom1, prom2, constant]); // all is a promise
+                // results is an array, the individual results correspond to the index of their
+                // promise as called in Promise.all()
+                all.then(function(results) {
+                    var prom1Result = results[0];
+                    var prom2Result = results[1];
+                    var constantResult = results[2];
+                });
+                return all;
+
+
+- waiting for a part of a promise chain, but not another part
+        If you have an asynchronous process that you want to wait to do something, but you also want to return to the caller before that something is done.
+        This is the equivalent of `return $.deferred.done()`
```

Comment:
$.Deferred

Reply:
but i don't know if it's worth mentioning it (you didn't for the previous examples)